data.md

```diff
@@ -3,9 +3,24 @@ WebAssembly Data
 
 ```k
 require "domains.k"
+```
+
+`WASM-SYNTAX` module is used to handle the syntax that is used only for parsing programs but not for parsing rules.
+In WebAssembly string is defined differently with K built-in strings, so we have to write the definition of WebAssembly `WasmString` in a separate module, and use the module just for parsing the program.
+Note that you cannot use a normal K `String` in any production definitions, because the K tokenizer does not support parsing ambiguity.
+
+```k
+module WASM-SYNTAX
```

Comment:
This should probably be called `WASM-STRING-SYNTAX` or something, and then in `wasm.md` we should have `WASM-SYNTAX` which is just 
```
module WASM-SYNTAX
  imports WASM-STRING-SYNTAX
  imports WASM
```

The way it's solved now works fine for this repo, but in Ewasm, for example (and other embedders) I want to import all of Wasm, without importing `WASM-TEST`. So the cleanest solution should be to have `WASM` and `WASM-SYNTAX` modules which do what you expect: define the entire syntax of Wasm, and the entire semantics.

Reply:
fixed

Reply:
... and maybe move the `WASM-SYNTAX` module to `wasm.md`, if that isn't already part of the upcoming refactoring.